### ISSUE TITLE
Questions about the offset 

### ISSUE THREAD

**Issue by Mehdishishehbor** [NamGyuKang/CosineSampler#3]
Hi,

Is there any reason the offset is diagonal? Was this with try and error?

Thanks,
Mehdi

**Comment by NamGyuKang** [comment]
Hello!

The diagonal was chosen because it was quick and easy to implement. After that, I tried a method of giving an offset in a circle around the point to be interpolate, but the result was not good due to an implementation problem. 

Alternatively, you can specify different offsets for the x-axis, y-axis, and z-axis and I did try and it worked. Using linspace can be made by adjusting the spacing, and you can choose things like rand or uniform.

Give it a try! :)

I think a circular offset rather than a diagonal shape is the way forward. 

Thank you very much for your continued interest. 
If you have any additional questions, don't hesitate to ask me! 

Have a good day.

Sincerely,
Namgyu Kang

**Comment by Mehdishishehbor** [comment]
Thank you for your quick response. I have some other questions:

- You mentioned in the paper that becasue of overfitting you are using multi-grid with the offset and if you don't use it the accuracy is not as good. However i have realized that if I use only one grid and offset only that grid to incorporate the next cells in the response the results wouldn't be as good as adding another grid and offseting. I am guessing if you were to use cubic interpolation your results wouldn't be as good as extra grids. 

- One other question. I don't see a nevier stokes (NS) PDE in the paper, but there is one in your repository. Have you tried different examples of NS with this approach? It seems NS is difficult to predict with this approach. Any idea why? Maybe PINNS works better for NS than PIXEL approach. what do you think?

**Comment by NamGyuKang** [comment]
Thank you for your questions.

I want to make sure my understanding is correct.

  - so you mean pixel with one grid & offset does not work well?
  

"grid to incorporate the next cells in the response the results" : 
  - could you explain more detail about "next cells"? I understand you used one grid (Multigrid=1). Are that next cells mean about (M, C, "H, W") or about other ("M", C, H, W)? (M=Multigrid, C=Channel, H=Height, W=Width) 
Or do you mean the out of range of some points due to the offset?

Umm we used multigrid because single grid's performance was not good for overfitting reason.
And I understand you tried with one grid and the results were also not good. I think I am missing something. Please letting me know if I miss.

- oh I uploaded the new version of paper in the arxiv this week which is the Camera-Ready of AAAI 2023. NS example can be found in NeurIPS workshop DLDE-2 spotlight paper, but I guess it was hard to find. Please take a look the new version of the paper pixel. 
( https://arxiv.org/pdf/2207.12800 )

I tried the multivariable inverse problem of NS.
For the forward problem, I tried hard to make it work, but the result was not good. And I guess the reason because of the curse of the dimensionality. Maybe more and more datapoints or multigrids increase the performance but I didn't try large scale (more than 500k points). In that time, I didn't have much time. But for NS training, I guess the offset's scale need to be changed maybe. I didn't try yet, but trying small scale of the   offset is worthy. I tried vorticity form of the NS equation forward and our code equation for the forward. I hope to solve the NS well with the future work with using pixel approach.

+ I added the 3d Helmholtz. You can see in the paper above.

For your information, I am now attending the AAAI 2023 for 1 weeks but I have enough time to respond. So if you have questions, please ask to me. I will reply as soon as possible. Thank you!!

Sincerely
Namgyu Kang

**Comment by Mehdishishehbor** [comment]
Thank you for your response. Great work and good luck with your presentation. One more question that I have at this point is the scaling factors of loss terms (multipliers behind bc loss, pdf loss and etc) . How did you find those values? try and error?

**Comment by NamGyuKang** [comment]
Thank you for your words about pixel and thank you for wishing me good luck with the presentation. 

I found the loss weighting values by dividing by 10 from 1 to 0.00001 and selected the one that learns well. 

And after I found something that worked, I checked the values ​​around that value.

Thank you very much.

I hope my comments were helpful. I will keep watching the github issue messages! 

Sincerely
Namgyu Kang.
